pkg/fake/ec2api.go

```diff
@@ -126,13 +128,23 @@ func (e *EC2API) CreateFleet(_ context.Context, input *ec2.CreateFleetInput, _ .
 			return nil, fmt.Errorf("missing launch template name")
 		}
 		var instanceIds []string
-		var skippedPools []CapacityPool
+		var icedPools []CapacityPool
+		var reservationExceededPools []CapacityPool
 		var spotInstanceRequestID *string
 
 		if string(input.TargetCapacitySpecification.DefaultTargetCapacityType) == karpv1.CapacityTypeSpot {
 			spotInstanceRequestID = aws.String(test.RandomName())
 		}
 
+		launchTemplates := map[string]*ec2.CreateLaunchTemplateInput{}
+		for e.CreateLaunchTemplateBehavior.CalledWithInput.Len() > 0 {
+			lt := e.CreateLaunchTemplateBehavior.CalledWithInput.Pop()
+			launchTemplates[*lt.LaunchTemplateName] = lt
+		}
+		for _, ltInput := range launchTemplates {
+			e.CreateLaunchTemplateBehavior.CalledWithInput.Add(ltInput)
```

Comment:
Same sort of question here -- behavior feels like it doesn't align with what the real EC2 behavior is